internal/backend/events/events.go

```diff
@@ -29,7 +29,10 @@ func (e *events) List(ctx context.Context, filter sdkservices.ListEventsFilter)
 }
 
 func (e *events) Save(ctx context.Context, event sdktypes.Event) (sdktypes.EventID, error) {
-	event = event.WithNewID().WithCreatedAt(time.Now())
+	event = event.WithNewID()
+	if event.CreatedAt() == time.Unix(0, 0).UTC() {
```

Comment:
I think this should assume this his set from outside and use what ever was provided

Reply:
I afraid that this will break the existing assumption that save event will set `created_at`, as it was before. That's why I kept setting created_at here if unset, just for sanity.

Moreover it's also misaligned with other sdktypes/db entities, for example session which is setting created_at in db layer. 
I suggest to open a ticket and decide what will  be the best option to handle it everywhere. It actually could be a part of your suggestion to add more `created_at` fields to other db entities. WDYT?

Reply:
Setting it here would create inconsistent and unreliable time measurements. Better not to set it here at all - then at least we can see if there's a problem that needs fixing (i.e. having events with an epoch timestamp).

As for an assumption that this function sets timestamps - I don't see that mentioned anywhere, so:
1. Return an error if the event doesn't have a real timestamp
2. Document this function and clarify in that comment that the function requires the caller to set the timestamp in advance

Alternatively, you can remove `WithCreatedAt` and add it as a parameter in `Save`.

Either way I agree that we need to standardize how we handle timestamps across all SDK types - please do open a ticket for that.

FYI, my personal preference is that it should be automatic without the caller having any say in the matter.

API consistency usually enforces measurement consistency, but to be clear measurement consistency in metrics is even more important than API consistency.

Reply:
Daniel, 
- in the current code the time actually is set *implicitly* *here*, in `Save()`. So for our purposes it was both consistent and reliable I think. I just added a check that event time could be already present in event passed to Save and if not set,  then event's time is updated from epoch to current time.
-  I don't want to change API (e.g. passing time to Save), not in this PR at least. We could discuss this at sync and see what's the best/consistent/convenient option to set event time. This PR tries to introduce a metric on top of existing API.
- I already tried several approaches and unfortunately none of them was convenient :( 
   -  Event is sdktype so every change to it creates new object, thus we cannot pass the reference to Event and update time on db layer before saving
   - we could change API such way that Save till return Event and not EventID, but it requires many changes in code. So I would like to keep this specific PR smaller and deal only with metrics and in other PR deal with where event time is set. WDYT?